jormungandr/src/settings/start/network.rs

```diff
@@ -107,24 +108,32 @@ impl From<super::config::TrustedPeer> for TrustedPeer {
 }
 
 impl Peer {
-    pub fn new(connection: SocketAddr, protocol: Protocol) -> Self {
+    pub fn new(connection: SocketAddr) -> Self {
+        Peer::with_timeout(
+            connection,
+            Duration::from_millis(DEFAULT_TIMEOUT_MILLISECONDS),
+        )
+    }
+
+    pub fn with_timeout(connection: SocketAddr, timeout: Duration) -> Self {
         Peer {
             connection,
-            protocol,
-            timeout: Duration::from_micros(DEFAULT_TIMEOUT_MICROSECONDS),
+            protocol: Protocol::Grpc,
+            timeout,
         }
     }
+
     pub fn address(&self) -> SocketAddr {
         self.connection
     }
 }
 
 impl Listen {
-    pub fn new(connection: SocketAddr, protocol: Protocol) -> Self {
+    pub fn new(connection: SocketAddr) -> Self {
         Listen {
             connection,
-            protocol,
-            timeout: Duration::from_micros(DEFAULT_TIMEOUT_MICROSECONDS),
+            protocol: Protocol::Grpc,
+            timeout: Duration::from_micros(DEFAULT_TIMEOUT_MILLISECONDS),
```

Comment:
you are still using `Duration::from_micros`
```suggestion
            timeout: Duration::from_millis(DEFAULT_TIMEOUT_MILLISECONDS),
```

Reply:
Thanks for noticing. I've used a const fn to initialize a `Duration` const in one place.